int8 y_axe
int8 x_axe
bool upside